test/FunctionalTests/MvcTests/CustomRoutingConventionTests.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using EntropyTests;
+using Microsoft.AspNet.Server.Testing;
+using Microsoft.AspNet.Testing.xunit;
+using Xunit;
+
+namespace BuilderTests.MvcTests
+{
+    public class CustomRoutingConventionTests
+    {
+        public const string SiteName = "Mvc.CustomRoutingConvention";
+
+        [ConditionalTheory]
+        [OSSkipCondition(OperatingSystems.Linux)]
+        [OSSkipCondition(OperatingSystems.MacOSX)]
+        [InlineData(ServerType.WebListener, RuntimeFlavor.Clr, RuntimeArchitecture.x86, "http://localhost:6200")]
+        [InlineData(ServerType.WebListener, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64, "http://localhost:6300")]
+        public async Task RunSite_WindowsOnly(
+            ServerType server,
+            RuntimeFlavor runtimeFlavor,
+            RuntimeArchitecture architecture,
+            string applicationBaseUrl)
+        {
+            await RunSite(server, runtimeFlavor, architecture, applicationBaseUrl);
+        }
+
+        [ConditionalTheory]
+        [OSSkipCondition(OperatingSystems.Windows)]
+        [InlineData(ServerType.WebListener, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64, "http://localhost:6400")]
+        [InlineData(ServerType.Kestrel, RuntimeFlavor.Mono, RuntimeArchitecture.x86, "http://localhost:6500")]
+        public async Task RunSite_Other(
+            ServerType server,
+            RuntimeFlavor runtimeFlavor,
+            RuntimeArchitecture architecture,
+            string applicationBaseUrl)
+        {
+            await RunSite(server, runtimeFlavor, architecture, applicationBaseUrl);
+        }
```

Comment:
I'm just aping the other functional tests in this repo. Do you know where I can find an example of testing routes in memory?